transcom/mymove
# [ MB-16408 ] Getting SIT Address Status updates from existing endpoint

## [Jira ticket](tbd)

## Summary

This change allows the Prime to request the entire Move and filter out
the JSON response to see the status of SIT Address Updates.

They can leverage a similar JQ expression like the following to see the
status of `MTOServiceItemDestSIT`.

```
jq '.mtoServiceItems[] | select(.modelType == "MTOServiceItemDestSIT")'
```

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Get a Move into a state where the Prime can ask for a SIT Address Update.
1. Make a SIT Address Update as the Prime.
1. Reject the SIT Address Update as the TOO.
1. Request the Move data from `/prime/v1/move-task-orders`.
1. See the `reason` and `status` in the `mtoServiceItems` response.

> A better **How to test** section may be provided. Reach to me on Slack for any
> help with this.
